STNDP-28 Intergrate Settings > Board name

changmoolee opened the pull request:
### Updated Result

![Mar-31-2025 20-51-04](https://github.com/user-attachments/assets/2cd267bc-cb9c-4deb-bc9d-57564092dec2)

- I used a server action to update the board name.
- Initially, I tried using react-hook-form, but it felt too heavy for managing just a single input.
Also, since I wanted to use a server action, I couldn’t use React Router’s <Form> component together with react-hook-form.

hyukkwonepic reviewed (changes requested):
# Please apply these suggestions when implementing the feature

## A
The action logic should be written in a separate route. In the routes.ts, there should be added route setting like this:
```ts
route(
  "/boards/:boardId/update",
  "routes/update-board-route/update-board-route.tsx"
),
```
And this route should only focus on updating the board resource.

  You can refer to `update-board-standup.tsx` file - where the route only responsible for the update action of the standup. The reason of doing is to separate responsibility of the action on each use cases by following the RESTful path.
    

## B

Please use `useFetcher` when updating the value.

You can refer to `todays-standup.tsx`'s `updateStandupFetcher` usage. For example:
```ts
<DynamicForm
  schema={schema}
  defaultValues={todayStandup.formData as DynamicFormValues}
  onSubmit={async (data) => {
    updateStandupFetcher.submit(
      {
        formData: data,
      },
      {
        encType: "application/json",
        method: "POST",
        action: `/boards/${board.id}/standups/${todayStandup?.id}/update`,
      }
    );
    setIsEditing(false);
  }}
  ...
```

hyukkwonepic commented:
NOTE
- I added some comments on the API side https://github.com/kiwinight/standup-kiwi/pull/17/commits/c078159fc78414c688de686520f00ea0b7b8c5cb

changmoolee commented:
Thanks for the feedback.

✅ **A**: Moved the action logic to a dedicated route (`update-board-route.tsx`) and added the corresponding route in `routes.ts`.

✅ **B**: Refactored to use `useFetcher` when updating the board name, following the pattern in `todays-standup.tsx`.

✅ **Missing Feature**: Disabled the save button after submission.

Let me know if there's anything else I should adjust.
